lib/ansible/module_utils/urls.py

```diff
@@ -1351,21 +1368,8 @@ def fetch_url(module, url, data=None, headers=None, method=None,
                      follow_redirects=follow_redirects, client_cert=client_cert,
                      client_key=client_key, cookies=cookies, use_gssapi=use_gssapi,
                      unix_socket=unix_socket)
-        # Lowercase keys, to conform to py2 behavior, so that py3 and py2 are predictable
-        info.update(dict((k.lower(), v) for k, v in r.info().items()))
```

Comment:
I believe we still need this `info.update` here. There are cases where `r.info()` is more than just headers iirc.

And I had completely forgotten that my `normalize_headers` didn't do anything with lowercasing.  So we'll need to add lowercasing back in.  Although I'd recommend not doing the lowercasing in `normalize_headers`.  I'd still do that on `info`, since it is a specific implementation detail of `fetch_url`.

Reply:
We can keep lowercase in `normalize_headers` function and add an extra argument to the function to activate lowercasing (`false` by default) ?